test/test.js

```diff
@@ -1427,6 +1428,21 @@ describe('node-fetch', () => {
 		});
 	});
 
+	// issue #414
+	it('should reject if attempt to accumulate body stream throws', function () {
+		let body = resumer().queue('a=1').end();
+		body = body.pipe(new stream.PassThrough());
+		const res = new Response(body);
+		const bufferConcat = Buffer.concat;
+		Buffer.concat = () => { throw new Error('embedded error'); };
+
+		return res.text().then(() => chai.assert(false)).catch(err => {
+			expect(err).to.be.an.instanceOf(FetchError)
+				.and.include({ type: 'system' })
+				.and.have.property('message').that.includes('Could not create Buffer')
+				.and.that.includes('embedded error');
+		}).then(() => Buffer.concat = bufferConcat);
```

Comment:
We need to reset `Buffer.concat` whether or not the test succeeded or failed.

Reply:
Good catch (no pun intended!). I think I changed some of the inner logic before submitting the PR and just forgot to update the final part of the chain. In any case, I've updated the PR now to reset in both cases. Thanks!